test/e2e_node/device_manager_test.go

```diff
@@ -384,15 +370,78 @@ var _ = SIGDescribe("Device Manager  [Serial] [Feature:DeviceManager][NodeFeatur
 				nodes, err := e2enode.TotalReady(ctx, f.ClientSet)
 				framework.ExpectNoError(err)
 				return nodes == 1
-			}, time.Minute, time.Second).Should(gomega.BeTrue())
+			}, 2*time.Minute, time.Second).Should(gomega.BeTrue())
 
 			// TODO: here we need to tolerate pods waiting to be recreated
 
 			ginkgo.By("making sure all the pods are ready after the recovery")
-			waitForPodConditionBatch(ctx, f, testPods, "Ready", 120*time.Second, testutils.PodRunningReady)
 
-			ginkgo.By("removing all the pods")
-			deleteBatch(ctx, f, testPods)
+			var devicePluginPodAfterRestart, tmpPod *v1.Pod
+
+			devicePluginPodAfterRestart, err = e2epod.NewPodClient(f).Get(ctx, devicePluginPod.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			err = e2epod.WaitForPodCondition(ctx, f.ClientSet, devicePluginPodAfterRestart.Namespace, testPod.Name, "Ready", 120*time.Second, testutils.PodRunningReady)
+			framework.ExpectNoError(err, "pod %s/%s did not go running", devicePluginPodAfterRestart.Namespace, devicePluginPodAfterRestart.Name)
+			framework.Logf("pod %s/%s running", devicePluginPodAfterRestart.Namespace, devicePluginPodAfterRestart.Name)
+
+			var devsLen int64
+			ginkgo.By("Waiting for the resource capacity/allocatable exported by the sample device plugin to become zero")
+			gomega.Eventually(func() bool {
+				node, ready := getLocalTestNode(ctx, f)
+				return ready &&
+					numberOfDevicesCapacity(node, resourceName) == devsLen &&
+					numberOfDevicesAllocatable(node, resourceName) == devsLen
+			}, 30*time.Second, framework.Poll).Should(gomega.BeTrue())
+
+			ginkgo.By("Checking that pod requesting devices failed to start because of admission error")
+			gomega.Eventually(ctx, func() bool {
```

Comment:
we should also probably spend few lines explaining what is the expected scenario is and how it unfolds. For example:
we now have exactly two pods running, device plugin pod and test pod. We can't control the recovery order.
if the test pod is recovered before the device plugin pod, great: that's exactly the scenario we want to exercise.
If the device plugin pod is recovered before the test pod, we still can meaningfully reproduce the scenario by NOT sending the registration command.

Reply:
Sure, I will explain this in more detail.